Surface_mesh_topology/include/CGAL/Surface_mesh_topology/internal/Shortest_noncontractible_cycle.h

```diff
@@ -0,0 +1,603 @@
+                     typename WeightFunctor::Weight_t* length,
+                     bool display_time=false)
+  { return compute_cycle(root_vertex, length, WeightFunctor(), display_time); }
+
+  template <class WeightFunctor=Unit_weight_functor>
+  Path compute_cycle(Original_dart_const_handle root_vertex,
+                     bool display_time=false)
+  { return compute_cycle(root_vertex, nullptr, display_time); }
+
+  template <class WeightFunctor>
+  Path compute_shortest_noncontractible_cycle(typename WeightFunctor::Weight_t* length,
+                                              const WeightFunctor& wf,
+                                              bool display_time=false)
+  {
+    CGAL::Timer t;
+    if (display_time)
+    { t.start(); }
+
+    m_cycle.clear();
+    bool first_check = true;
+    typename WeightFunctor::Weight_t min_length=0;
+    for (auto it=get_local_map().template attributes<0>().begin(),
+           itend=get_local_map().template attributes<0>().end(); it!=itend; ++it)
+    {
+      Dart_handle dh=get_local_map().template dart_of_attribute<0>(it);
+      typename WeightFunctor::Weight_t temp_length;
+      if (first_check)
+      {
+        if (compute_cycle(dh, m_cycle, &temp_length, nullptr, wf))
+        {
+          min_length = temp_length;
+          first_check = false;
+        }
+      }
+      else
+      {
+        if (compute_cycle(dh, m_cycle, &temp_length, &min_length, wf))
+        { min_length = temp_length; }
+      }
+    }
+    if (length!=nullptr) { *length=min_length; }
+
+    if (display_time)
+    {
+      t.stop();
+      std::cout<<"[TIME] compute_shortest_noncontractible_cycle: "<<t.time()<<" seconds."<<std::endl;
+    }
+
+    return m_cycle;
+  }
+
+  template <class WeightFunctor=Unit_weight_functor>
+  Path compute_shortest_noncontractible_cycle(typename WeightFunctor::Weight_t* length,
+                                              bool display_time=false)
+  { return compute_shortest_noncontractible_cycle(length, WeightFunctor(), display_time); }
+
+  Path compute_edgewidth(bool display_time=false)
+  { return compute_shortest_noncontractible_cycle(nullptr, display_time); }
+
+protected:
+  int vertex_info(Dart_handle dh) const
+  { return get_local_map().template info<0>(dh); }
+  int& vertex_info(Dart_handle dh)
+  { return get_local_map().template info<0>(dh); }
+
+  void create_vertex_info()
+  {
+    for (auto it=get_local_map().darts().begin(),
+           itend=get_local_map().darts().end(); it!=itend; ++it)
+    {
+      if (get_local_map().template attribute<0>(it)==nullptr)
+      { get_local_map().template set_attribute<0>
+          (it, get_local_map().template create_attribute<0>(-1)); }
+    }
+  }
+
+  void initialize_vertex_info()
+  {
+    for (auto it=get_local_map().template attributes<0>().begin(),
+           itend = get_local_map().template attributes<0>().end(); it != itend; ++it)
+    { get_local_map().template info_of_attribute<0>(it)=-1; }
```

Comment:
@gdamiand  are you also using this strange indentation style are is it only due to the new author?

Reply:
If needed I can run a astyle on the files to fix those. (it will also fix the extra indentation due to namespace)

Reply:
Done.